src/main/java/seedu/address/model/schedule/Event.java

```diff
@@ -99,6 +99,29 @@ public static boolean isValidEvent(Event event) {
         return EventDescription.isValidEventDescription(event.getEventDescription().toString());
     }
 
+    /**
+     * Compares 2 {@code Event} based on date and time. Returns a positive integer if {@code event}
+     * occurs after the caller, a negative integer if {@code event} occurs before the caller, and 0
+     * if both {@code Event} have occurs on the same date and time.
+     */
+    public int compareTo(Event event) {
+        LocalDate eventDate = event.getDate();
+        if (date.isBefore(eventDate)) {
+            return -1;
+        } else if (date.isAfter(eventDate)) {
+            return 1;
+        }
+
+        LocalTime eventTime = event.getTime();
+        if (time.isBefore(eventTime)) {
+            return -1;
+        } else if (time.isAfter(eventTime)) {
+            return 1;
+        } else {
+            return 0;
```

Comment:
should we further compare the duration in this case?

Reply:
That would be possible. Should the longer event be first or the shorter event?

Reply:
i think the shorter event should be first, as they finish first